android-components/components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabViewHolder.kt

```diff
@@ -108,14 +108,12 @@ class DefaultTabViewHolder(
         updateSelectedTabIndicator(isSelected)
 
         // In the final else case, we have no cache or fresh screenshot; do nothing instead of clearing the image.
-        if (thumbnailLoader != null && tab.content.thumbnail == null) {
+        if (thumbnailLoader != null) {
             val thumbnailSize = THUMBNAIL_SIZE.dpToPx(thumbnailView.context.resources.displayMetrics)
             thumbnailLoader.loadIntoView(
                 thumbnailView,
                 ImageLoadRequest(id = tab.id, size = thumbnailSize),
             )
-        } else if (tab.content.thumbnail != null) {
-            thumbnailView.setImageBitmap(tab.content.thumbnail)
```

Comment:
This check corresponds to the case when `thumbnailLoader` is `null`. What is the UI/UX state there when this case happens if we remove this check and do nothing for `(thumbnailLoader == null)` case?

Reply:
Good question! In this case, the `thumbnailLoader` wouldn't do anything and the tabs tray item would render nothing just a square. UX-wise this isn't great, however, the `TabViewHolder` implementation is no longer used in anywhere exception the sample-browser and reference-browser.

Fenix uses a custom implementation of the `TabViewHolder` that fallsback to rendering the favicon if a thumbnail is unavailable.

Once Fenix moves to a Jetpack Compose tabs tray and no longer depends on the API hierarchy of the `TabViewHolder` we can remove this code/component entirely and the sample/reference browsers can also move to a Compose UI. There were some thoughts about unifying the implementations under a common tabs tray, but with Compose, that seems to be a little less valuable.

You can read a bit about this under the Compose section of the Android Components blog post: https://mozac.org/2021/07/05/whats-next.html

Reply:
Got it! Thanks for the details and the link. I will be digging into that. 😄